### ISSUE TITLE
Why records property is not updating with fetch?

### ISSUE THREAD

**Issue by DmitryFillo** [jeromegn/Backbone.localStorage#177]
I use localStorage collection, works great. But I want use storage event for updating collection in many tabs. Idea: do fetch() when storage event fires.

```
bindStorageEvent : function() {
    var _this = this;
    $(window).on('storage.'+this.localStorage.name, function(e) {
        if(e.originalEvent.key === _this.localStorage.name) {
            _this.fetch();
        }
    });
},

unbindStorageEvent : function() {
    $(window).off('storage.'+this.localStorage.name);
}
```

But this shouldn't work as expetcted because [records property](https://github.com/jeromegn/Backbone.localStorage/blob/master/backbone.localStorage.js#L73) will not be updated with fetch and [save() method use records prop](https://github.com/jeromegn/Backbone.localStorage/blob/master/backbone.localStorage.js#L80). And so all actual data which came from one tab will be rewrited by old data from another tab.

So I rewrited code in this way:

```
bindStorageEvent : function() {
    var _this = this;
    $(window).on('storage.'+this.localStorage.name, function(e) {
        if(e.originalEvent.key === _this.localStorage.name) {

            // the dirtiest records update way, only for test
            var store = _this.localStorage.localStorage().getItem(_this.localStorage.name);
            _this.localStorage.records = (store && store.split(",")) || [];

            _this.fetch();
        }
    });
},

unbindStorageEvent : function() {
    $(window).off('storage.'+this.localStorage.name);
}
```

And it works great.

Suggested solution: I think that we can put this code to the another method in the localStorage object (records_update) and call this method from constructor and from Backbone.sync() when doing fetch() or from findAll() before getting particular model. I think this property must be updated on fetch, not only at the collection initialization time.

**Comment by scott-w** [comment]
So if I'm understanding it correctly, every time you run `fetch()`, the in-memory model should update with whatever's in the `localStorage`, overwriting the data in the model/collection. I'll pull a test together to see if I can verify this behaviour.

**Comment by scott-w** [comment]
I'm removing the `2.0` milestone until I can understand what needs to happen here. I'm thinking it _might_ be a bug but I'm not sure.

**Comment by Chris136** [comment]
I have the same problem.
If I replace "this.records" in the function "findAll" with "localStorage.getItem(this.name).split(',')" everything is working.
model.LocalStorage and the browser localStorage seem to be out of sync.

**Comment by scott-w** [comment]
Hi @Chris136 - is the local storage being updated in multiple places? e.g. are there two models referencing the local storage record?

**Comment by Chris136** [comment]
No, just one model. I use it to save calendar events created via drag and drop. I noticed newly created events were missing when I switched a week back and forth. But after a complete site reload, the events were back where they were created.
After this little fix I mentioned before, everything worked as intended.

**Comment by scott-w** [comment]
Thanks for the information @Chris136 If I understand this correctly, what's happening is:

1. Fetch all events from localStorage
2. Add new event
3. Move away from view
4. Move back to view
5. Step 1. fires again but the newly added item in localStorage isn't in your view

Basically, once there's a list of records in the collection, `fetch` is just returning that list instead of checking the localStorage again.

I can have a look at this over the weekend and try to re-create the issue in a test environment.

**Comment by scott-w** [comment]
Hi @Chris136 

Would it be possible for you to check the version of Backbone.localStorage that you're using? I've been trying to recreate this in tests and the output seems to be acting the way I'd expect in version 2.0.0.

**Comment by Chris136** [comment]
@scott-w according to the changelog I use version 1.1.16 (January 22, 2015).
I installed it via NPM @2.x and it said version 2.0.0

**Comment by scott-w** [comment]
Thanks @Chris136, is the issue still present?

**Comment by Chris136** [comment]
@scott-w yes, that is the version where I noticed the problem.

**Comment by Chris136** [comment]
I try to make a minimal test case if I find the time this week.

**Comment by scott-w** [comment]
HI @Chris136 

That'll be awesome, thanks 👍

**Comment by yahuarkuntur** [comment]
Any updates on this issue? or is there any workaround?

Cheers!

**Comment by blikblum** [comment]
I fixed this in my backbone fork.

Look at the following commits:
https://github.com/blikblum/backbone/commit/8054cdd740eb7d62489185ffb055ab9f97e0a7a1
https://github.com/blikblum/backbone/commit/e183cce5dc262ca91559c94230134db03ad40de2

Should not be hard to make a PR. Currently i'm on a slow internet connection so cannot do it myself
